src/main/java/com/google/api/codegen/config/LongRunningConfig.java

```diff
@@ -176,6 +182,57 @@ private static LongRunningConfig createLongRunningConfigFromProtoFile(
         totalPollTimeout);
   }
 
+  /** Creates an instance of LongRunningConfig from a Discovery method. */
+  @Nullable
+  static LongRunningConfig createLongRunningConfig(DiscoveryMethodModel methodModel) {
+
+    com.google.api.codegen.discovery.Method method = methodModel.getDiscoMethod();
+
+    for (String part : method.id().split("\\.")) {
+      if (DISCOVERY_LRO_TYPE.toLowerCase().equals(part.toLowerCase())) {
+        // This is a method from the Operation Client itself; it isn't a method that *uses* LRO.
+        return null;
+      }
+    }
+    if (method.id().toLowerCase().contains(DISCOVERY_LRO_TYPE.toLowerCase())) {
+      return null;
+    }
+
+    // LRO methods will have Operation as the return type.
+    try {
+      if (method.response() == null
+          || Strings.isNullOrEmpty(method.response().reference())
+          || !DISCOVERY_LRO_TYPE.equals(method.response().reference())) {
+        return null;
+      }
+    } catch (NullPointerException e) {
+      System.exit(1);
+    }
+
+    // Use default retry settings.
+    Duration initialPollDelay = Duration.ofMillis(LRO_INITIAL_POLL_DELAY_MILLIS);
```

Comment:
Default retry settings will soon be removed from client configuration anyway.

Reply:
What about non-default?

Reply:
I think the new philosophy is to not expose user configuration unless it's necessary and there is need for it. That philosophy has driving the API config schema simplification effort, and I, because I'm lazy, also do not want to expose retry setting toggles.